### PR TITLE
docs: add templates for move to googleapis/python-bigquery-sqlalchemy repo

### LINKED ISSUE #85: Add noxfile.py to test with multiple versions of Python and our dependencies
We use [nox](https://nox.thea.codes/en/stable/) in the `google-cloud-bigquery` client library ([source](https://github.com/googleapis/python-bigquery/blob/master/noxfile.py)) to test on multiple versions of Python as well as multiple versions of our dependencies.

This would be useful for issues like https://github.com/mxmzdlv/pybigquery/issues/83 where we want to add support for SQLAlchemy 1.4.0 without breaking support for earlier versions.

### PR DESCRIPTION
**Pull request by tswast**
Closes #85

### CODE REVIEW

File: synth.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 Google LLC
```
Comment (tswast) [inline synth.py:1]: Manual code

File: setup.py [hunk 2]
```diff
@@ -21,39 +21,58 @@
 import io
 from setuptools import setup
 
+# Package metadata.
```
Comment (tswast) [inline setup.py:24]: Manual code

File: setup.cfg [hunk 3]
```diff
@@ -1,21 +1,19 @@
...
-
-[tool:pytest]
-addopts= --tb short --capture no --color=yes --flake8
-python_files=test/*test_*.py
-
-[sqla_testing]
-requirement_cls=pybigquery.requirements:Requirements
-profile_file=.profiles.txt
-
-[db]
```
Comment (tswast) [inline setup.cfg:19]: I might need this?

**Comment by tswast** [comment]
Test output (local)

```
$ nox
nox > Running session unit-3.6
nox > Creating virtual environment (virtualenv) using python3.6 in .nox/unit-3-6
nox > pip install mock pytest pytest-cov -c /Users/swast/src/pybigquery/testing/constraints-3.6.txt
nox > pip install -e . -c /Users/swast/src/pybigquery/testing/constraints-3.6.txt
nox > py.test --quiet --junitxml=unit_3.6_sponge_log.xml --cov=pybigquery --cov=tests/unit --cov-append --cov-config=.coveragerc --cov-report= --cov-fail-under=0 tests/unit
.....................................                                                                                       [100%]
----------------------------- generated xml file: /Users/swast/src/pybigquery/unit_3.6_sponge_log.xml -----------------------------


37 passed in 1.14s
nox > Session unit-3.6 was successful.
nox > Running session unit-3.7
nox > Creating virtual environment (virtualenv) using python3.7 in .nox/unit-3-7
nox > pip install mock pytest pytest-cov -c /Users/swast/src/pybigquery/testing/constraints-3.7.txt
nox > pip install -e . -c /Users/swast/src/pybigquery/testing/constraints-3.7.txt
nox > py.test --quiet --junitxml=unit_3.7_sponge_log.xml --cov=pybigquery --cov=tests/unit --cov-append --cov-config=.coveragerc --cov-report= --cov-fail-under=0 tests/unit
.....................................                                                                                       [100%]
----------------------------- generated xml file: /Users/swast/src/pybigquery/unit_3.7_sponge_log.xml -----------------------------


37 passed in 1.17s
nox > Session unit-3.7 was successful.
nox > Running session unit-3.8
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/unit-3-8
nox > pip install mock pytest pytest-cov -c /Users/swast/src/pybigquery/testing/constraints-3.8.txt
nox > pip install -e . -c /Users/swast/src/pybigquery/testing/constraints-3.8.txt
nox > py.test --quiet --junitxml=unit_3.8_sponge_log.xml --cov=pybigquery --cov=tests/unit --cov-append --cov-config=.coveragerc --cov-report= --cov-fail-under=0 tests/unit
.....................................                                                                                       [100%]
----------------------------- generated xml file: /Users/swast/src/pybigquery/unit_3.8_sponge_log.xml -----------------------------


37 passed in 1.09s
nox > Session unit-3.8 was successful.
nox > Running session unit-3.9
nox > Creating virtual environment (virtualenv) using python3.9 in .nox/unit-3-9
nox > pip install mock pytest pytest-cov -c /Users/swast/src/pybigquery/testing/constraints-3.9.txt
nox > pip install -e . -c /Users/swast/src/pybigquery/testing/constraints-3.9.txt
nox > py.test --quiet --junitxml=unit_3.9_sponge_log.xml --cov=pybigquery --cov=tests/unit --cov-append --cov-config=.coveragerc --cov-report= --cov-fail-under=0 tests/unit
.....................................                                                                                       [100%]
----------------------------- generated xml file: /Users/swast/src/pybigquery/unit_3.9_sponge_log.xml -----------------------------


37 passed in 1.54s
nox > Session unit-3.9 was successful.
nox > Running session system-3.8
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/system-3-8
nox > pip install --pre grpcio
nox > pip install mock pytest google-cloud-testutils -c /Users/swast/src/pybigquery/testing/constraints-3.8.txt
nox > pip install -e . -c /Users/swast/src/pybigquery/testing/constraints-3.8.txt
nox > py.test --quiet --junitxml=system_3.8_sponge_log.xml tests/system
........................................                                                                                    [100%]
======================================================== warnings summary =========================================================
tests/system/test_sqlalchemy_bigquery.py::test_engine_with_dataset
tests/system/test_sqlalchemy_bigquery.py::test_dataset_location
tests/system/test_sqlalchemy_bigquery.py::test_reflect_select
tests/system/test_sqlalchemy_bigquery.py::test_custom_expression
  /Users/swast/src/pybigquery/.nox/system-3-8/lib/python3.8/site-packages/google/cloud/bigquery/client.py:444: UserWarning: Cannot create BigQuery Storage client, the dependency google-cloud-bigquery-storage is not installed.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
---------------------------- generated xml file: /Users/swast/src/pybigquery/system_3.8_sponge_log.xml ----------------------------
40 passed, 4 warnings in 54.32s
nox > Session system-3.8 was successful.
nox > Running session cover
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/cover
nox > pip install coverage pytest-cov
nox > coverage report --show-missing --fail-under=50
Name                                Stmts   Miss Branch BrPart  Cover   Missing
-------------------------------------------------------------------------------
pybigquery/__init__.py                  0      0      0      0   100%
pybigquery/api.py                      16     16      2      0     0%   20-43
pybigquery/parse_url.py               112      1     50      2    98%   41, 232->242
pybigquery/sqlalchemy_bigquery.py     331    219    104      0    26%   60, 72, 82-83, 90-108, 111-121, 154-157, 162-164, 174-175, 181-209, 216-218, 221, 228, 231, 234, 237, 240, 243, 246, 249, 256, 260, 263-270, 273-285, 319-324, 328, 333, 338-343, 348-362, 370-411, 419, 422-441, 446-462, 467-516, 519-529, 532-536, 543-558, 561-584, 587-588, 594, 598, 601-619, 622-629, 632-635, 638-641, 645, 649, 653
tests/unit/test_dialect_types.py        6      0      2      0   100%
tests/unit/test_parse_url.py           62      0      8      0   100%
-------------------------------------------------------------------------------
TOTAL                                 527    236    166      2    50%
nox > coverage erase
nox > Session cover was successful.
nox > Running session lint
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/lint
nox > pip install flake8 black==19.10b0
nox > black --check docs pybigquery tests noxfile.py setup.py
All done! ✨ 🍰 ✨
12 files would be left unchanged.
nox > flake8 pybigquery tests
tests/unit/test_parse_url.py:201:83: E231 missing whitespace after ','
tests/system/test_sqlalchemy_bigquery.py:72:36: E231 missing whitespace after ','
tests/system/test_sqlalchemy_bigquery.py:73:49: E231 missing whitespace after ','
tests/system/test_sqlalchemy_bigquery.py:220:37: E231 missing whitespace after ','
tests/system/test_sqlalchemy_bigquery.py:389:75: E231 missing whitespace after ','
nox > Command flake8 pybigquery tests failed with exit code 1
nox > Session lint failed.
nox > Running session lint_setup_py
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/lint_setup_py
nox > pip install docutils pygments
nox > python setup.py check --restructuredtext --strict
running check
nox > Session lint_setup_py was successful.
nox > Running session blacken
nox > Creating virtual environment (virtualenv) using python3.6 in .nox/blacken
nox > pip install black==19.10b0
nox > black docs pybigquery tests noxfile.py setup.py
All done! ✨ 🍰 ✨
12 files left unchanged.
nox > Session blacken was successful.
nox > Running session docs
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/docs
nox > pip install -e .
nox > pip install sphinx alabaster recommonmark
nox > sphinx-build -W -T -N -b html -d docs/_build/doctrees/ docs/ docs/_build/html/
Running Sphinx v3.5.3
making output directory... done
[autosummary] generating autosummary for: multiprocessing.rst
loading intersphinx inventory from https://python.readthedocs.org/en/latest/objects.inv...
loading intersphinx inventory from https://googleapis.dev/python/google-auth/latest/objects.inv...
loading intersphinx inventory from https://googleapis.dev/python/google-api-core/latest/objects.inv...
loading intersphinx inventory from https://grpc.github.io/grpc/python/objects.inv...
loading intersphinx inventory from https://proto-plus-python.readthedocs.io/en/latest/objects.inv...
intersphinx inventory has moved: https://python.readthedocs.org/en/latest/objects.inv -> https://python.readthedocs.io/en/latest/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: [new config] 1 added, 0 changed, 0 removed
reading sources... [100%] multiprocessing


Traceback (most recent call last):
  File "/Users/swast/src/pybigquery/.nox/docs/lib/python3.8/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/Users/swast/src/pybigquery/.nox/docs/lib/python3.8/site-packages/sphinx/application.py", line 352, in build
    self.builder.build_update()
  File "/Users/swast/src/pybigquery/.nox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 296, in build_update
    self.build(to_build,
  File "/Users/swast/src/pybigquery/.nox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 310, in build
    updated_docnames = set(self.read())
  File "/Users/swast/src/pybigquery/.nox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 420, in read
    raise SphinxError('master file %s not found' %
sphinx.errors.SphinxError: master file /Users/swast/src/pybigquery/docs/index.rst not found

Sphinx error:
master file /Users/swast/src/pybigquery/docs/index.rst not found
nox > Command sphinx-build -W -T -N -b html -d docs/_build/doctrees/ docs/ docs/_build/html/ failed with exit code 2
nox > Session docs failed.
nox > Ran multiple sessions:
nox > * unit-3.6: success
nox > * unit-3.7: success
nox > * unit-3.8: success
nox > * unit-3.9: success
nox > * system-3.8: success
nox > * cover: success
nox > * lint: failed
nox > * lint_setup_py: success
nox > * blacken: success
nox > * docs: failed
```

I can fix up those trailing commas in this PR for the lint session, but for docs I'd prefer to cover that in a follow-up PR. https://github.com/googleapis/python-bigquery-sqlalchemy/issues/90

**Review by shollyman — APPROVED** [review]
Templated and manual files seem okay.

Should there be something added in this or another PR indicating the transitions in the CHANGELOG or README?  Or address that when you cut the first release?

**Comment by tswast** [comment]
> Should there be something added in this or another PR indicating the transitions in the CHANGELOG or README? Or address that when you cut the first release?

Good call. Filed https://github.com/googleapis/python-bigquery-sqlalchemy/issues/91 to do this.

I'm hoping that release-please can take over ownership of the changelog, so I'll mention the move in the commit subject for this PR.

**Pull request merged**